clicmd

```diff
@@ -1,12 +1,68 @@
 #!/bin/bash -eu
 
+#
+# YADRO OpenBMC Command Line Interface
+# Copyright (C) 2020-2021 YADRO
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
 # Base CLI command handler.
 # This script is used for setting up the environment, elevating privileges
 # and handling hierarchical commands.
 
 # prevent modification of executable path
 PATH="/bin:/sbin:/usr/bin:/usr/sbin"
 
+# Role related definitions
+ROLE_ADMIN="admin"
+ROLE_OPERATOR="operator"
+ROLE_USER="user"
+
+GROUP_ADMIN="%GROUP_ADMIN%"
+GROUP_OPERATOR="%GROUP_OPERATOR%"
+GROUP_USER="%GROUP_USER%"
+
+# If we are running escalated, we have GROUP already set by the caller
+if [ "${USER}" != "root" -o ! -v GROUP ]; then
+  if [[ "${USER}" = "root" ]]; then
+    echo "Can't run CLI commands as root directly, use an admin account"
+    exit 1
+  fi
+  GROUP=$(id -Gn | sed -n 's/.*\(%GROUP_REGEX%\)[ ]*.*$/\1/gp')
+fi
+
+# Make it available to the escalated invocation
+export GROUP
```

Comment:
I didn't get it. Do you want to pass it through `sudo` or is it just for subscripts?

Reply:
Yes, it should pass through `sudo` and without `export` it won't. Verified.

Reply:
Normally, sudo should ignore any exported variables.
If you pass GROUP from user session to sudo's one and sudo sees GROUP - it is a bug in our setup.
Now user can export own PATH, PAGER, etc. It may ends up, for example, with executing /tmp/busctl instead of /usr/bin/busctl.

Reply:
> If you pass GROUP from user session to sudo's one and sudo sees GROUP - it is a bug in our setup.

As I can see only the GPOUP variable will be passed.
The `install.sh` makes it at the line 127.
So, I don't see anything wrong in this way.

> Yes, it should pass through `sudo` and without `export` it won't. Verified.

To be a more paranoid I would try to fix the line 110 to something like:
```bash
exec GPROUP=${GROUP} sudo ${USER_CMD} ${sub_path//_/ } "$@"
```
instead of using export. ;)

Reply:
Yes, I didn't notice GROUP in the installation script.

Reply:
> ```shell
> exec GPROUP=${GROUP} sudo ${USER_CMD} ${sub_path//_/ } "$@"
> ```

I checked. The GROUP declaration should be before `exec`.
```bash
GPROUP=${GROUP} exec sudo ${USER_CMD} ${sub_path//_/ } "$@"
```